Jumpscale/clients/blockchain/tfchain/TFChainCapacity.py

```diff
@@ -63,7 +63,75 @@ def _grid_broker_pub_key(self):
             self._grid_broker_pub_key_ = vk.to_curve25519_public_key()
         return self._grid_broker_pub_key_
 
-    def reserve_s3(self, email, threebot_id, location, size=1, source=None, refund=None):
+    def _notary_data_get(self, hash_key):
+        data = self._notary_client.get(hash_key)
+        sk = self._threebot_singing_key(data.threebot_id)
+        pk = _signing_key_to_private_key(sk)
+        box = nacl.public.Box(pk, self._grid_broker_pub_key)
+        data_dict = j.data.serializers.msgpack.loads(box.decrypt(data.content))
+
+        return data.threebot_id, data_dict
+
+    def reservations_transactions_list(self):
```

Comment:
I think this can be improved.
With the current code, every time you want to list the reservation made, we have to hit both the explorers and the notary.
We could improve that by saving the transaction ids of the reservation in bcdb.

So instead of walking on all the transaction every time, we could have something like:
- adapt the schema of the object and add a transaction id list
- reservations_transactions_list first check if the transaction id list is empty
- if empty do what the current code does, but also save the transaction id
- if not, just return the transactions from the list

With this logic, after we release this version, the user will only do the full walk of their transaction once. Then all the next time he will only get the id from bcdb which saves a lot of calls on the notary.